OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs

```diff
@@ -18,38 +18,40 @@ namespace OpenRA.Mods.Common.Warheads
 {
 	public class HealthPercentageDamageWarhead : DamageWarhead
 	{
-		[Desc("Size of the area. Damage will be applied to this area.", "If two spreads are defined, the area of effect is a ring, where the second value is the inner radius.")]
-		public readonly WDist[] Spread = { new WDist(43) };
+		[Desc("Size of the area. Damage will be applied to this area.")]
+		public readonly WDist Spread = new WDist(43);
 
 		public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageModifiers)
 		{
 			var world = firedBy.World;
 
 			var debugVis = world.WorldActor.TraitOrDefault<DebugVisualizations>();
 			if (debugVis != null && debugVis.CombatGeometry)
-				world.WorldActor.Trait<WarheadDebugOverlay>().AddImpact(pos, Spread, DebugOverlayColor);
-
-			var range = Spread[0];
-			var hitActors = world.FindActorsInCircle(pos, range);
-			if (Spread.Length > 1 && Spread[1].Length > 0)
-				hitActors = hitActors.Except(world.FindActorsInCircle(pos, Spread[1]));
+			{
+				WDist[] spread = { Spread };
+				world.WorldActor.Trait<WarheadDebugOverlay>().AddImpact(pos, spread, DebugOverlayColor);
+			}
 
+			// FindActors*On*Circle is needed to include actors who are close enough that their HitShape might intersect with spread border.
+			var hitActors = world.FindActorsOnCircle(pos, Spread);
 			foreach (var victim in hitActors)
-				DoImpact(victim, firedBy, damageModifiers);
-		}
-
-		public override void DoImpact(Actor victim, Actor firedBy, IEnumerable<int> damageModifiers)
-		{
-			if (!IsValidAgainst(victim, firedBy))
-				return;
-
-			var healthInfo = victim.Info.TraitInfoOrDefault<IHealthInfo>();
-			if (healthInfo == null)
-				return;
-
-			// Damage is measured as a percentage of the target health
-			var damage = Util.ApplyPercentageModifiers(healthInfo.MaxHP, damageModifiers.Append(Damage, DamageVersus(victim)));
-			victim.InflictDamage(firedBy, new Damage(damage, DamageTypes));
+			{
+				if (!IsValidAgainst(victim, firedBy))
+					continue;
+
+				var closestSpreadPos = WPos.Lerp(pos, victim.CenterPosition, Spread.Length, (victim.CenterPosition - pos).Length);
```

Comment:
If `Spread.Length` is bigger than `(victim.CenterPosition - pos).Length` this will lerp past the center of the actor and out the other side!

I'll push a fix to this alongside a few other cleanups.